### PR TITLE
Fix OAuth flow

### PR DESCRIPTION
**Pull request by DiodonHystrix**
In OAuth `oauth_token_secret` has to be passed to get an access token.
It works with Twitter, because Twitter doesn't follow spec. At least, that's the information I found.

I will try to wrap it up tomorrow. (I need to add more tests.)

### CODE REVIEW

**Comment by danschultzer** [comment]
Looks good, thanks!

I think we should remove both `state` and `oauth_token_secret` from the plug methods and controller and make them just accept a map called e.g. `session_params` (I'm not sure if this is a good name) that the strategies define. This way, we can simplify the controller and plug, and no changes are necessary to add some other strategy that's not OAuth based, but still needs some temporarily stored params. Wdyt?

**Comment by DiodonHystrix** [comment]
Perfect! Was thinking about similar solution. :)

**Comment by danschultzer** [comment]
I've rebased with master, and it went from 130 to 71 additions!

Also FYI I've removed the `: socket_dir` config setting since I use mac with brew postgres setup and it couldn't connect to the db.

**Comment by danschultzer** [comment]
Oh it's even code as there's no longer a need for the OAuth test provider since the session params in plug and controllers are generalized! Feel free to merge this into master whenever you have tested and ensured it works with your OAuth setup.

File: lib/pow_assent/strategies/oauth.ex [hunk 1]
```diff
@@ -76,7 +76,8 @@ defmodule PowAssent.Strategy.OAuth do
     credentials      = OAuther.credentials([
       consumer_key: config[:consumer_key],
       consumer_secret: config[:consumer_secret],
-      token: oauth_token])
+      token: oauth_token,
+      token_secret: config[:session_params][:oauth_token_secret]])
```
Comment (DiodonHystrix) [inline lib/pow_assent/strategies/oauth.ex]: I just added `[:session_params]` because it was missing.

**Comment by DiodonHystrix** [comment]
Feel free to finish it, because I won't be available in the next few days.
Anyway, thanks a lot for help Dan :)

**Comment by danschultzer** [comment]
Yeah, I'll finish it up 😄 I got distracted yesterday, but almost got it done with proper tests.

**Comment by danschultzer** [comment]
I've added proper tests for OAuth now, so it verifies the signature and verifier in the bypass callback. @DiodonHystrix can you check if this PR works as expected with your OAuth setup?

**Comment by DiodonHystrix** [comment]
I will check the PR tomorrow again and hopefully we will merge it :)

**Comment by DiodonHystrix** [comment]
I've checked it again and read the PR. All seems to be fine. :)

**Pull request merged**